rotkehlchen/chain/ethereum/modules/the_graph/decoder.py

```diff
@@ -0,0 +1,179 @@
+        stake_amount = token_normalized_value(
+            token_amount=hex_or_bytes_to_int(context.tx_log.data[:32]),
+            token=token,
+        )
+        # find Transfer event with delegation tax (0.5% GRT burn) and persist the burnt amount
+        tokens_burnt = 0
+        for ctx_log in context.all_logs:
+            if (
+                ctx_log.topics[0] == TOPIC_TRANSFER and
+                hex_or_bytes_to_address(ctx_log.topics[1]) == CONTRACT_THE_GRAPH_STAKING and
+                hex_or_bytes_to_address(ctx_log.topics[2]) == ZERO_ADDRESS
+            ):
+                tokens_burnt = hex_or_bytes_to_int(ctx_log.data)
+        # identify and override the original stake Transfer event
+        for event in context.decoded_events:
+            if (
+                event.location_label == delegator and
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                event.event_type = HistoryEventType.STAKING
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Delegated {stake_amount} GRT to Indexer {indexer}'
+                if tokens_burnt > 0:
+                    taxed_amount = token_normalized_value(
+                        token_amount=tokens_burnt,
+                        token=token,
+                    )
+                    event.notes += f', {taxed_amount} GRT has been burnt as delegation tax'
+                event.counterparty = CPT_THE_GRAPH
+        return DEFAULT_DECODING_OUTPUT
+
+    # processes StakeDelegatedLocked event
+    def _decode_stake_locked(
+            self,
+            token: 'EvmToken',
+            context: DecoderContext,
+    ) -> DecodingOutput:
+        delegator = hex_or_bytes_to_address(context.tx_log.topics[2])
+        if delegator is None or self.base.is_tracked(delegator) is False:
+            return DEFAULT_DECODING_OUTPUT
+        indexer = hex_or_bytes_to_address(context.tx_log.topics[1])
+        tokens_amount = hex_or_bytes_to_int(context.tx_log.data[:32])
+        lock_timeout_secs = hex_or_bytes_to_int(context.tx_log.data[64:128])
+        tokens_amount_norm = token_normalized_value(
+            token_amount=tokens_amount,
+            token=token,
+        )
+        # create a new informational event about undelegation and the expiring lock on tokens
+        event = self.base.make_event_from_transaction(
+            transaction=context.transaction,
+            tx_log=context.tx_log,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=A_GRT,
+            balance=Balance(),
+            location_label=delegator,
+            notes=(
+                f'Undelegated {tokens_amount_norm} GRT'
+                f' from Indexer {indexer}'
+                f', lock expires in {lock_timeout_secs} seconds'
```

Comment:
The reason I didn't use `,` in the end of line: https://github.com/rotki/rotki/issues/5432#issuecomment-1750724659

Reply:
no worries. I explained why in discord. It's unrelated to that issue.